pkg/ccl/streamingccl/streamproducer/replication_stream_test.go

```diff
@@ -367,3 +368,90 @@ CREATE TABLE t2(
 		}
 	})
 }
+
+func TestStreamAddSSTable(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	h, cleanup := streamingtest.NewReplicationHelper(t, base.TestServerArgs{}, serverutils.TestTenantID())
+	defer cleanup()
+
+	h.Tenant.SQL.Exec(t, `
+CREATE DATABASE d;
+CREATE TABLE d.t1(i int primary key, a string, b string);
+INSERT INTO d.t1 (i) VALUES (42);
+USE d;
+`)
+
+	ctx := context.Background()
+	rows := h.SysDB.QueryStr(t, "SELECT crdb_internal.start_replication_stream($1)", h.Tenant.ID.ToUint64())
+	streamID := rows[0][0]
+
+	const streamPartitionQuery = `SELECT * FROM crdb_internal.stream_partition($1, $2)`
+
+	dataSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "GET" {
+			if _, err := w.Write([]byte("42,42\n")); err != nil {
+				t.Logf("failed to write: %s", err.Error())
+			}
+		}
+	}))
+	defer dataSrv.Close()
+
+	testAddSSTable := func(t *testing.T, initialScan bool, addSSTableBeforeRangefeed bool, table string) {
+		var startTime time.Time
+		h.Tenant.SQL.Exec(t, fmt.Sprintf("CREATE TABLE %s (id INT PRIMARY KEY, n INT)", table))
+		h.Tenant.SQL.QueryRow(t, "SELECT clock_timestamp()").Scan(&startTime)
+		startHlcTime := hlc.Timestamp{WallTime: startTime.UnixNano()}
+		if initialScan {
+			startHlcTime = hlc.Timestamp{}
+		}
+		if addSSTableBeforeRangefeed {
+			h.Tenant.SQL.Exec(t, fmt.Sprintf("IMPORT INTO %s CSV DATA ($1)", table), dataSrv.URL)
+		}
+		source, feed := startReplication(t, h, makePartitionStreamDecoder,
+			streamPartitionQuery, streamID, encodeSpec(t, h, startHlcTime, table))
+		defer feed.Close(ctx)
+		if !addSSTableBeforeRangefeed {
+			h.Tenant.SQL.Exec(t, fmt.Sprintf("IMPORT INTO %s CSV DATA ($1)", table), dataSrv.URL)
+		}
+
+		codec := source.codec.(*partitionStreamDecoder)
+		for {
+			require.True(t, source.rows.Next())
+			source.codec.decode()
+			if codec.e.Batch != nil {
```

Comment:
How do you know if addSSt requests were issued? Looks like data server produces just 1 line? that will not result in addsst requests, afaik.